### PR TITLE
[Docs] Improve channel-list components documentation

### PR DESCRIPTION
**Pull request by evsaev**
### 🎯 Goal

Improve the documentation for both LLC/UI channel-list controllers.

### 📝 Summary

- Update documentation for `ChatChannelListController`, bring special attention to `filter` closure 
- Explain how to show `ChatChannelListVC` modally/in the navigation controller/in tab bar/as a child 

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

**Review by bielikb — APPROVED** [review]

File: docusaurus/docs/iOS/uikit/controllers/channels.md [hunk 1]
```diff
@@ -2,38 +2,108 @@
...
-Given the nature of `Filter`, having generic logic capable of using `query.filter` at runtime to determine if a ChatChannel meets the criteria is extremely expensive and complex. Instead, having a block that only checks for the particular needs of the use case is much cheaper.
+An integrator should make sure to keep a strong reference to the `delegate` passed to the controller. Otherwise, the delegate object will get deallocated since the controller references it **weakly**.
 :::
 
+### 4. Synchronize controller
+
+The `synchronize` should be called on the controller to:
+- fetch the first page of channels matching the query
+- subscribe on events for those channels
+- setup the database observers
```
Comment (nuno-vieira) [inline docusaurus/docs/iOS/uikit/controllers/channels.md:97]: The database observers are an implementation detail of the controller, and so the customers don't know, and shouldn't know about this. What about "start observing data changes", WDYT? 
```suggestion
The `synchronize` should be called on the controller to:
- fetch the first page of channels matching the query
- subscribe to events for those channels
- start observing data changes
```

**Review by nuno-vieira — CHANGES_REQUESTED** [review]
Great documentation 👏 We really needed this! Really liked the structure, it is really easy to follow 👌 

Requesting changes just because of some typos, and also to fix some grammar issues which Grammerly helped me spotting 🙂

File: docusaurus/docs/iOS/uikit/controllers/channels.md [hunk 2]
```diff
@@ -2,38 +2,108 @@
...
+```swift
+let filter1: Filter<ChannelListFilterScope> = .equal(.type, to: .messaging)
 ```
-client.channelListController(query: query, filter: { channel in
-    // Your filtering logic
+as well as conditions for **custom** fields:
+```swift
+let filter2: Filter<ChannelListFilterScope> = .equal("state", to: "LA")
+```
+Primitive filters can be **combined** using `and/or/nor` operators which allows to get a filter of any complexity:
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/controllers/channels.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/controllers/channels.md [hunk 3]
```diff
@@ -2,38 +2,108 @@
...
-)
+// 3. Set the delegate
+controller.delegate = delegate
+
+// 4. Synchronize a controller
+controller.synchronize { error in /* handle error */ }
 ```
 
-### Having more than one ChannelListController active at the same time
+Let's go though step by step.
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/controllers/channels.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/components/channel-list.md [hunk 4]
```diff
@@ -6,30 +6,76 @@ import SingletonNote from '../../common-content/chat-client.md'
 import ComponentsNote from '../../common-content/components-note.md'
 import Properties from '../../common-content/reference-docs/stream-chat-ui/chat-channel-list/chat-channel-list-vc-properties.md'
 
-This component is used to display a list of channels. Channels are fetched from the API and kept in sync automatically. Behind the scenes this component uses the `Query Channels` API endpoint and Websocket events.
+The `ChatChannelListVC` is the UI component used to display a list of channels matching the given query.
+
+:::note
+Channel list screen is backed by the low-level `ChatChannelListController` component which fetches channels from the API and keeps the list in sync with the remote.
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/components/channel-list.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/components/channel-list.md [hunk 5]
```diff
@@ -6,30 +6,76 @@ import SingletonNote from '../../common-content/chat-client.md'
...
+The first step to show the channel list screen in your app is to create `ChatChannelListVC` instance:
+```swift
+// Create `ChatChannelController` with the given query
+let controller = ChatClient.shared.channelListController(query: ..., filter: ...)
+
+// Create a view controller backed by the low-level controller
+let channelListVC = ChatChannelListVC.make(with: controller)
+```
+
+When the `ChatChannelListVC` instance is created, there are be multiple ways of showing it:
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/components/channel-list.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/components/channel-list.md [hunk 6]
```diff
@@ -6,30 +6,76 @@ import SingletonNote from '../../common-content/chat-client.md'
...
+        navigationVC.didMove(toParent: self)
     }
 }
 ```
 
-<SingletonNote />
-
-If you have a navigation controller setup the SDK will navigate from the channel list to the channel view controller when the user taps on a channel. In the example above we are showing the channels where the current user is a member, you can change the query to fit your application needs.
+:::note
+In all the cases but the push to existed navigation controller, channel list is wrapped by `UINavigationController` before presenting to get the `channel list -> channel` navigation out of the box.
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/components/channel-list.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/controllers/channels.md [hunk 7]
```diff
@@ -2,38 +2,108 @@
...
 
+import SingletonNote from '../../common-content/chat-client.md'
+
 ## ChannelListController
 
-`ChatChannelListController` allows you to observe a list of chat channels based on the provided query and to paginate channels.
+`ChatChannelListController` is the component responsible for managing the list of channels matching the given query. The main responsibilities are:
+- exposing the list of channels matching the query
+- allowing to paginate the channel list (initially, only the 1st page of channels is fetched)
+- keeping the list of channels in sync with remote by dynamically linking/unlinking channels that start/stop matching the query
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/controllers/channels.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/controllers/channels.md [hunk 8]
```diff
@@ -2,38 +2,108 @@
...
+```
+Primitive filters can be **combined** using `and/or/nor` operators which allows to get a filter of any complexity:
+```swift
+let compoundFilter: Filter<ChannelListFilterScope> = .and([
+    filter1,
+    filter2
+])
+```
+
+The `query.sort` is an array of sorting options. Sorting option are applied based on their order in array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
```
[the remark this answers is not in the document]
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/controllers/channels.md]: Fixed ✅

File: docusaurus/docs/iOS/uikit/controllers/channels.md (hunk 1 above)
Comment (evsaev) [reply docusaurus/docs/iOS/uikit/controllers/channels.md]: Fixed ✅

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**